pkg/syncer/metadatasyncer.go

```diff
@@ -2980,3 +2985,95 @@ func startStoragePolicyQuotaCRInformer(ctx context.Context, cfg *restclient.Conf
 	}()
 	return nil
 }
+
+// storagePolicyUsageCRSync patches StoragePolicyUsage CRs for k8s PVs in Bound state
+func storagePolicyUsageCRSync(ctx context.Context, metadataSyncer *metadataSyncInformer) {
```

Comment:
- Can you add logic to create StoragePolicyUsage CR if it does not exist for a bound PVC?
- Can you also add logic to create StoragePolicyUsage CR if it does not exist for an unbound PVC? (regarding this one, you can submit a separate PR if that's easier)

Reply:
I have added the ask # 1 to create the SP Usage CR.
Testing:

Deleted the SP Usage CR:
```
kubectl delete storagepolicyusage -n storage-policy-test wcp-profile-wkaoxja3iq-pvc-usage
storagepolicyusage.cns.vmware.com "wcp-profile-wkaoxja3iq-pvc-usage" deleted


# kubectl get storagepolicyusage -n storage-policy-test wcp-profile-wkaoxja3iq-pvc-usage -o yaml
Error from server (NotFound): storagepolicyusages.cns.vmware.com "wcp-profile-wkaoxja3iq-pvc-usage" not found
```
Let the full sync invoke the CR sync method added by this PR, and verified that the CR gets created as expected.
Total used capacity: 1650Mi
```
kubectl get pvc -n storage-policy-test
NAME             STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS             AGE
example-pvc-10   Bound    pvc-e0f64f22-4c33-42de-8b7d-780c0fb2a96d   110Mi      RWO            wcp-profile-wkaoxja3iq   3h35m
example-pvc-11   Bound    pvc-f766bfe2-52d6-40ec-bc86-efe14effd49b   110Mi      RWO            wcp-profile-wkaoxja3iq   3h35m
example-pvc-12   Bound    pvc-7b05d732-2c7a-4f13-931c-6adf6c38f738   110Mi      RWO            wcp-profile-wkaoxja3iq   3h35m
example-pvc-13   Bound    pvc-827a36c0-beb2-4370-99f9-80a13164afcb   110Mi      RWO            wcp-profile-wkaoxja3iq   3h35m
example-pvc-14   Bound    pvc-9f276148-a795-4a2e-adcc-84b2e8e5c02f   110Mi      RWO            wcp-profile-wkaoxja3iq   3h35m
example-pvc-15   Bound    pvc-c10d5b68-de1f-41f3-b6bc-35892879a86c   110Mi      RWO            wcp-profile-wkaoxja3iq   3h35m
example-pvc-16   Bound    pvc-b4b12b71-0c3a-4d80-bf3b-711a63591bd1   110Mi      RWO            wcp-profile-wkaoxja3iq   3h35m
example-pvc-2    Bound    pvc-ffd068e5-e620-4cd6-82c9-d79e0d17148d   110Mi      RWO            wcp-profile-wkaoxja3iq   3h35m
example-pvc-3    Bound    pvc-bbe0746f-97a2-490f-9055-aedc18e3fcac   110Mi      RWO            wcp-profile-wkaoxja3iq   3h35m
example-pvc-4    Bound    pvc-74d50572-71cf-4d5c-9896-8eed3bef2cc4   110Mi      RWO            wcp-profile-wkaoxja3iq   3h35m
example-pvc-5    Bound    pvc-0561628b-b174-4a04-adcb-d4808445e0b3   110Mi      RWO            wcp-profile-wkaoxja3iq   3h35m
example-pvc-6    Bound    pvc-6b39297a-dc08-416f-acee-11856e5bad47   110Mi      RWO            wcp-profile-wkaoxja3iq   3h35m
example-pvc-7    Bound    pvc-ebbfd0d8-222e-480c-b0e2-e304ee0383c3   110Mi      RWO            wcp-profile-wkaoxja3iq   3h35m
example-pvc-8    Bound    pvc-48116413-fb1d-4482-b769-8d5b0aab0f55   110Mi      RWO            wcp-profile-wkaoxja3iq   3h35m
example-pvc-9    Bound    pvc-97321596-ef2c-477e-a6b3-cff4a5033f82   110Mi      RWO            wcp-profile-wkaoxja3iq   3h35m
```

```
kubectl get storagepolicyusage -n storage-policy-test wcp-profile-wkaoxja3iq-pvc-usage -o yaml                   4216522683360f78652aa4c30ece5339: Fri Dec 29 02:38:19 2023

apiVersion: cns.vmware.com/v1alpha1
kind: StoragePolicyUsage
metadata:
  creationTimestamp: "2023-12-29T02:38:06Z"
  generation: 17
  name: wcp-profile-wkaoxja3iq-pvc-usage
  namespace: storage-policy-test
  resourceVersion: "2606889"
  uid: a7cd2667-b53c-48c2-83b5-f6d64f9f5e70
spec:
  resourceApiGroup: ""
  resourceExtensionName: volume.cns.vsphere.vmware.com
  resourceKind: PersistentVolumeClaim
  storageClassName: wcp-profile-wkaoxja3iq
  storagePolicyId: c268ab52-ee7f-409b-b524-241ac2e5d2f9
status: 
  quotaUsage:
    reserved: "0"
    used: 1650Mi
```

Reply:
I will add the ask # 2 in a separate PR